packages/mattermost-redux/src/reducers/entities/cloud.ts

```diff
@@ -64,11 +64,21 @@ function invoices(state: Record<string, Invoice> | null = null, action: GenericA
     }
 }
 
-const emptyLimits = Object.freeze({});
-export function limits(state: Limits = emptyLimits, action: GenericAction) {
+export interface LimitsReducer {
+    limits: Limits;
+    limitsLoaded: boolean;
+}
+const emptyLimits = Object.freeze({
```

Comment:
question here? why explicit need to freeze the object?

Reply:
Out of habit of enforcing immutability. If any code inadvertently mutates the object, an exception will be thrown, which will encourage the developer to white their code in a non-mutating way.

Reply:
This is the first usage of `Object.freeze` in the codebase, besides the freezing of the redux store while in `DeveloperMode`. I'm not against using it, but what constitutes it here, and should it be used elsewhere as well? I imagine there's other cases where nested state is being used in a reducer, like this case.

Reply:
Since we already freeze the whole store in developer mode, this is redundant and can be removed. The store wide one is better by:
* being a deep freeze of nested properties
* only happening in dev mode, so not incurring perf penalties in prod mode.
* not needing to be placed in all the reducer files in the codebase